compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -479,32 +514,53 @@ LogicalResult insertLogicalObjectFifoAccess(ModuleOp moduleOp) {
       }
     });
 
+    // We maintain a map from AllocOp to LogicalObjectFifoAccessOp in order to
+    // avoid creating a new LogicalObjectFifoAccessOp for the same AllocOp being
+    // used in a different op.
+    DenseMap<Value, AMDAIE::LogicalObjectFifoAccessOp>
+        memrefToLogicalObjectFifoAccess;
     WalkResult res = coreOp->walk([&](Operation *op) {
-      if (auto linalgOp = dyn_cast<linalg::LinalgOp>(op)) {
-        for (auto &&[idx, operand] :
-             llvm::enumerate(linalgOp->getOpOperands())) {
-          if (memrefToLogicalObjectFifo.contains(operand.get())) {
+      if (isa<linalg::LinalgOp, vector::TransferReadOp,
+              vector::TransferWriteOp>(op)) {
+        for (auto &&[idx, operand] : llvm::enumerate(op->getOpOperands())) {
+          if (memrefToLogicalObjectFifoAccess.contains(operand.get())) {
+            op->setOperand(idx, memrefToLogicalObjectFifoAccess[operand.get()]);
+          } else if (memrefToLogicalObjectFifo.contains(operand.get())) {
             rewriter.setInsertionPointToStart(coreOp.getBody());
             std::tuple<AMDAIE::LogicalObjectFifoFromMemrefOp,
                        AMDAIE::MemoryAccess>
                 value = memrefToLogicalObjectFifo[operand.get()];
             auto accessOp = rewriter.create<AMDAIE::LogicalObjectFifoAccessOp>(
                 rewriter.getUnknownLoc(), std::get<0>(value),
                 std::get<1>(value));
-            linalgOp->setOperand(idx, accessOp);
-          } else if (auto type =
-                         llvm::dyn_cast<MemRefType>(operand.get().getType())) {
+            memrefToLogicalObjectFifoAccess[operand.get()] = accessOp;
+            op->setOperand(idx, accessOp);
+          } else if (operand.get().getDefiningOp() &&
```

Comment:
Why do we need to specifically check alloc ops?

I.e. why did you replace:

```
auto type = llvm::dyn_cast<MemRefType>(operand.get().getType())
```

with:

```
operand.get().getDefiningOp() && isa<memref::AllocOp>(operand.get().getDefiningOp())
```
?

The former looks more general?

Reply:
Forgot to revert to the previous condition check. Thanks for pointing that out!